model/credentials/db.go

```diff
@@ -13,49 +14,64 @@ import (
 )
 
 const (
-	Collection  = "credentials"
-	CAName      = "evergreen"
-	ServiceName = "evergreen.mongodb.com"
+	Collection = "credentials"
+	CAName     = "evergreen"
 
-	defaultConnectionTimeout = 10 * time.Second
-	defaultExpiration        = 30 * 24 * time.Hour // 1 month
+	defaultExpiration = 30 * 24 * time.Hour // 1 month
+)
+
+var (
+	serviceName        string
+	errNotBootstrapped = errors.Errorf("%s collection has not been bootstrapped", Collection)
 )
 
 // Bootstrap performs one-time initialization of the credentials collection with
-// the certificate authority and evergreen service certificate.
-func Bootstrap() error {
-	env := evergreen.GetEnvironment()
+// the certificate authority and service certificate. In order to perform
+// operations on this collection, collection, this must succeed.
+func Bootstrap(env evergreen.Environment) error {
 	settings := env.Settings()
-	maxExpiration := time.Duration(1<<63 - 1)
+	if settings.DomainName == "" {
+		return errors.Errorf("bootstrapping %s collection requires domain name to be set in admin settings", settings.DomainName)
+	}
 
-	caConfig := certdepot.CertificateOptions{
+	maxExpiration := time.Duration(math.MaxInt64)
+	serviceConfig := certdepot.CertificateOptions{
 		CA:         CAName,
-		CommonName: CAName,
+		CommonName: settings.DomainName,
+		Host:       settings.DomainName,
 		Expires:    maxExpiration,
 	}
 
-	serviceConfig := certdepot.CertificateOptions{
+	caConfig := certdepot.CertificateOptions{
 		CA:         CAName,
-		CommonName: ServiceName,
-		Host:       ServiceName,
+		CommonName: CAName,
 		Expires:    maxExpiration,
 	}
 
 	bootstrapConfig := certdepot.BootstrapDepotConfig{
 		MongoDepot:  mongoConfig(settings),
 		CAName:      CAName,
 		CAOpts:      &caConfig,
-		ServiceName: ServiceName,
+		ServiceName: settings.DomainName,
 		ServiceOpts: &serviceConfig,
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), defaultConnectionTimeout)
+	ctx, cancel := env.Context()
 	defer cancel()
 
 	if _, err := certdepot.BootstrapDepotWithMongoClient(ctx, env.Client(), bootstrapConfig); err != nil {
-		return errors.Wrap(err, "could not bootstrap credentials collection")
+		return errors.Wrapf(err, "could not bootstrap %s collection", Collection)
 	}
 
+	serviceName = settings.DomainName
+
+	return nil
+}
+
+func validateBootstrapped() error {
```

Comment:
to avoid this we could have a certdepot method in the environment, and use dependency injection to get out of the need for the global